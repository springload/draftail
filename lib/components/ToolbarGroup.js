import PropTypes from 'prop-types';
import React from 'react';

const ToolbarGroup = ({ children }) => {
    const hasChildren = React.Children.toArray(children).some(c => c !== null);
    return hasChildren ? (
        <div className="Draftail-ToolbarGroup">{children}</div>
    ) : null;
};

ToolbarGroup.propTypes = {
    children: PropTypes.node,
};

ToolbarGroup.defaultProps = {
    children: null,
};

export default ToolbarGroup;
