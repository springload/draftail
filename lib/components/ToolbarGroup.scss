.#{$NS}ToolbarGroup {
    display: inline;
    padding-right: $controls-spacing / 2;

    & + & {
        border-left: 1px solid $color-editor-chrome-accent;
        padding-left: $controls-spacing / 2;
    }
}
