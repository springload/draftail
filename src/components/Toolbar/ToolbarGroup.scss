@use "../../api/constants" as *;

.#{$DRAFTAIL}ToolbarGroup {
  display: inline-block;

  &::before {
    content: "";
    display: inline-block;
    width: 0.0625rem;
    height: 1rem;
    vertical-align: middle;
    margin: 0 $controls-spacing * 0.5;
    background-color: $draftail-editor-chrome-accent;
  }

  &:first-of-type::before {
    display: none;
  }

  .#{$DRAFTAIL}Editor--readonly & {
    opacity: $draftail-editor-readonly-opacity;
  }
}
