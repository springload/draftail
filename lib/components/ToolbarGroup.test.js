import React from "react";
import { shallow } from "enzyme";
import ToolbarGroup from "./ToolbarGroup";

describe("ToolbarGroup", () => {
  it("children", () => {
    expect(
      shallow(
        <ToolbarGroup>
          <span>Test child</span>
        </ToolbarGroup>,
      ),
    ).toMatchSnapshot();
  });

  it("empty", () => {
    expect(shallow(<ToolbarGroup />)).toMatchSnapshot();
  });

  it("empty children", () => {
    expect(
      shallow(
        <ToolbarGroup>
          {null}
          {null}
        </ToolbarGroup>,
      ),
    ).toMatchSnapshot();
  });

  it("empty array children", () => {
    expect(shallow(<ToolbarGroup>{[]}</ToolbarGroup>)).toMatchSnapshot();
  });
});
