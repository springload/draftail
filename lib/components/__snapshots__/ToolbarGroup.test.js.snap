// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarGroup children 1`] = `
<div
  className="Draftail-ToolbarGroup"
>
  <span>
    Test child
  </span>
</div>
`;

exports[`ToolbarGroup empty 1`] = `""`;

exports[`ToolbarGroup empty array children 1`] = `""`;

exports[`ToolbarGroup empty children 1`] = `""`;
